web3.js/src/connection.ts

```diff
@@ -1958,33 +2041,15 @@ export type SignatureSubscriptionCallback = (
  * Signature subscription options
  */
 export type SignatureSubscriptionOptions = {
-  commitment?: Commitment;
+  commitment?: Commitment; // Must default to `Connection._commitment` or 'finalized' otherwise
```

Comment:
Is this comment directed at maintainers? If so, I think we should move this comment (and the others you've copied around) to the `args: IWSRequestParams` param on `_makeSubscription`. I don't think that the current comment makes it very clear why it must default to some commitment, but that explanation should be read by contributor who calls `_makeSubscription` in a new function. In fact, it could generally describe that the client should set explicit defaults for parameters to ensure that subscriptions are hashed correctly.